src/inputs/input/input.jsx

```diff
@@ -29,46 +50,133 @@ const propTypes = {
+     */
     icon: PropTypes.oneOfType([
         PropTypes.shape({}),
         PropTypes.string,
     ]),
+    /**
+     * The id is used for the a label\'s for.
+     */
     id: PropTypes.string,
+    /**
+     * Format to appear on dark backgrounds.
+     */
     inverse: PropTypes.bool,
+    /**
+     * Changes the general behavior of the Text Mask component.
+     */
     keepCharPositions: PropTypes.bool,
+    /**
+     * Optional Label to display with the Input.
+     */
     label: PropTypes.string,
+    /**
+     * Position the label above or below the input.
+     */
     labelPosition: PropTypes.oneOf(['bottom', 'top']),
+    /**
+     * Supply any inline styles to the label.
+     */
     labelStyle: PropTypes.shape({}),
+    /**
+     * An icon input field can show that it is currently loading data.
+     */
     loading: PropTypes.bool,
+    /**
+     * Is an array that defines how the user input is going to be masked.
+     */
     mask: PropTypes.oneOfType([
         PropTypes.array,
         PropTypes.func,
     ]),
+    /**
+     * Specifies the maximum value the field can have.
+     */
     max: PropTypes.number,
+    /**
+     * Specifies the maximum number of characters that the user can enter.
+     */
     maxLength: PropTypes.number,
+    /**
+     * Specifies the minimum value the field can have.
+     */
     min: PropTypes.number,
+    /**
+     * Specifies the minimum number of characters that the user needs to enter.
+     */
     minLength: PropTypes.number,
+    /**
+     * Specifies name of the field
+     */
     name: PropTypes.string,
+    /**
+     * Can handle an onBlur event from parent.
+     */
     onBlur: PropTypes.func,
+    /**
+     * Can handle an onChange event from parent.
+     */
     onChange: PropTypes.func,
+    /**
+     * Can handle an onClick event from parent.
+     */
     onClick: PropTypes.func,
+    /**
+     * Can handle an onFocus event from parent.
+     */
     onFocus: PropTypes.func,
+    /**
+     * Can handle an onKeyDown event from parent.
+     */
     onKeyDown: PropTypes.func,
+    /**
+     * A hint to the user of what can be entered in the input.
+     */
     placeholder: PropTypes.string,
+    /**
+     * Specifies that the user must fill in a value before submitting a form.
+     */
     required: PropTypes.bool,
+    /**
+     * Specifies whether or not to show the spinners for the numeric control. This attribute is optional.
+     */
     showSpinners: PropTypes.bool,
+    /**
+     * Supply any inline styles to the Input\'s container. Mainly used for padding and margins.
+     */
     style: PropTypes.shape({}),
+    /**
+     * An Input can receive focus.
+     */
     tabIndex: PropTypes.oneOfType([
         PropTypes.number,
         PropTypes.string,
     ]),
+    /**
+     * The HTML input type.
+     */
     type: PropTypes.oneOf(['email', 'number', 'password', 'tel', 'text']),
+    /**
+     * The initial value of the control. This attribute is optional.
```

Comment:
Maybe add some text here to indicate you have to use this if using `onChange` and using the `<Input>` as a controlled component.
```
The initial value of the control. This attribute is optional; however, you must use it
if using `onChange` prop and using the Input as a controlled component.
```

Reply:
Looks good. Resolving.